conans/client/tools/files.py

```diff
@@ -369,6 +370,23 @@ def dos2unix(filepath):
     _replace_with_separator(filepath, "\n")
 
 
+def rename(src, dst):
+    """
+    rename a file or folder to avoid "Access is denied" error on Windows
+    :param src: Source file or folder
+    :param dst: Destination file or folder
+    """
+    if platform.system() == "Windows" and which("robocopy") and os.path.isdir(src):
+        if os.path.exists(dst):
+            raise ConanException("rename {} to {} failed, dst exists.".format(src, dst))
+        completed = subprocess.run(["robocopy", "/move", "/e", "/ndl", "/nfl", src, dst],
+                                   stdout=subprocess.PIPE)
+        if completed.returncode != 1:
+            raise ConanException("rename {} to {} failed.".format(src, dst))
+    else:
+        os.rename(src, dst)
```

Comment:
We should return the same type of exception if the `rename` fails. I'm ok if we capture all different errors and raise a `ConanException('rename failed....')`.

Reply:
@jgsogo Could you be more specific? Both exceptions are same type of ConanException with different error messages,right?